server.js

```diff
@@ -0,0 +1,138 @@
+import express from "express";
+import bodyParser from "body-parser";
+import mongoose, { ObjectId } from "mongoose";
+const app = express();
+const PORT = 3001;
+const connectionString =
+  "mongodb+srv://Prohar:PEA16april2005@cluster0.bosgtkc.mongodb.net/?retryWrites=true&w=majority";
+
+const URL = {
+  GREETINGS: "/api/greetings",
+  NOTES: "/api/notes",
+};
+
+app.use(express.json());
+
+// DB CONNECTION
+mongoose
+  .connect(connectionString, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+  })
+  .then(() => {
+    console.log("Connected to MongoDB");
+  })
+  .catch((error) => {
+    console.error("Error connecting to MongoDB:", error.message);
+  });
+
+// SCHEMA
+const schema = {
+  id: Number,
+  title: String,
+  content: String,
+  createAt: Date,
+  updateAt: Date,
+};
+const mongooseModel = mongoose.model("NEWCOL", schema);
+
+// TASK1
+app.get(URL.GREETINGS, function (req, res) {
+  try {
+    const name = req.query.name;
+    if (name) {
+      res.send(`hello ${name}`);
+    } else {
+      throw new Error("there is no value");
+    }
+  } catch (error) {
+    console.log(error);
+  }
+});
+
+// GET
+app.get(URL.NOTES, async (req, res) => {
+  res.send({});
+});
+
+// POST
+app.post(URL.NOTES, async (req, res) => {
+  console.log("inside post function");
+
+  const data = new mongooseModel({
+    id: req.body.id,
+    title: req.body.title,
+    content: req.body.content,
+    createAt: Date.now(),
+    updateAt: "",
+  });
+
+  const val = await data.save();
```

Comment:
All the logic referenced to the calculations/catching data located into controllers. Already described above. But all the logic referenced to work with DB should be described into `src/db/providers`. Example: src/db/providers/notesProvider.js